infrastructures/docker/environment/src/main/java/org/eclipse/che/workspace/infrastructure/docker/environment/dockerimage/EntryPointParser.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.docker.environment.dockerimage;
+
+import static com.fasterxml.jackson.databind.node.JsonNodeType.STRING;
+import static java.lang.String.format;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.eclipse.che.api.core.model.workspace.config.MachineConfig;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.commons.annotation.Nullable;
+
+/**
+ * Can be used to parse container entry-point definition specified as a YAML list of strings or a
+ * single YAML string.
+ */
+public final class EntryPointParser {
```

Comment:
what's the reason to be final class?

Reply:
According to Effective Java, classes should either be designed to be extended or final. This class is not designed for extension - it has a single simple purpose, so no need for it to be extended. Composition over inheritance ;)

Reply:
According to reality, I removed `final` in a few classes because it prevents from creating mock for unit tests.
Since this object is a simple POJO, It definitely is not a case for creating mock. But still I would not add final here since all other our POJOs are not final as well.